bspump/declarative/abc.py

```diff
@@ -52,7 +52,10 @@ def walk(self, parent=None, key=None):
 
 		yield (parent, key, self)
 
-		for key in self.Attributes:
+		# Attributes may change during iteration, which could produce an error
+		current_attributes = self.Attributes.copy()
```

Comment:
Change of attributes is likely an error. When this happens?

Reply:
@ateska When in f. e. !WHEN you dynamically add a new attribute such as self.Then0, you change the self.Attributes, which may throw the error.

Reply:
(walk is also used during initialization as well as optimization)